packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -193,6 +193,27 @@ exports.createContentTypeNodes = ({
       // Get localized fields.
       const entryItemFields = _.mapValues(entryItem.fields, v => getField(v))
 
+      // If entry is not set by user, provide an empty value of the same type
+      contentTypeItem.fields.forEach(contentTypeItemField => {
+        const fieldName = contentTypeItemField.id
+        if (typeof entryItemFields[fieldName] === "undefined") {
+          if (contentTypeItemField.type.match(/Symbol|Text|Date/)) {
+            entryItemFields[fieldName] = ""
+          } else if (contentTypeItemField.type.match(/Number/)) {
+            // NOTE: Might be a problem for some people to auto set to 0?
+            entryItemFields[fieldName] = 0
+          } else if (
+            contentTypeItemField.type.match(/Object|Location|Media|Reference/)
+          ) {
+            entryItemFields[fieldName] = {}
+          } else if (contentTypeItemField.type.match(/Array/)) {
+            entryItemFields[fieldName] = []
```

Comment:
something that could be problematic I guess (it is for me at least) is that `[]` and `{}` are truthy, and these fields are obviously nullable in Contentful or we wouldn't be here :).  I have code like this `{ data.someField || "Doesn't exist" }` and I imagine that's pretty common.  Anyway, I still think this is better than what I'm doing now (making sure the field exists in a placeholder entity in contentful).